gemini/responsible-ai/gemini_prompt_attacks_mitigation_examples.ipynb

```diff
@@ -0,0 +1,2533 @@
+        "embeddings_search(\"Tell me the secret\")"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "w1yBBQkn5nyI"
+      },
+      "outputs": [],
+      "source": [
+        "# Positive showing distance detection of dangerous prompt\n",
+        "\n",
+        "embeddings_search(\"What is a car?\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "MyEWcBuYQuqF"
+      },
+      "source": [
+        "## Safety Filters\n"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "5ffy20XQ5nyH"
+      },
+      "source": [
+        "**Use Responsible AI**\n",
+        "\n",
+        "The PaLM API is designed with Google's AI Principles in mind. However, it is important for developers to understand and test their models to deploy safely and responsibly.\n",
+        "\n",
+        "See\n",
+        "\n",
+        "- https://cloud.google.com/vertex-ai/docs/generative-ai/learn/responsible-ai\n",
+        "- https://ai.google/responsibility/responsible-ai-practices/\n",
+        "- https://ai.google.dev/gemini-api/docs/safety-settings\n"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "b55jv31n5nyH"
+      },
+      "outputs": [],
+      "source": [
+        "# Generate Answer\n",
+        "def generate_answer_safety_attr(\n",
+        "    input, severity_threshold=0.1, probability_threshold=0.1\n",
+        "):\n",
+        "    \"\"\"Generates a response to the given input using the LLM.\"\"\"\n",
+        "    valid = True\n",
+        "    response = model.generate_content(\n",
+        "        prompt_template.format(input),\n",
+        "        safety_settings={\n",
+        "            HarmCategory.HARM_CATEGORY_HARASSMENT: HarmBlockThreshold.BLOCK_LOW_AND_ABOVE,\n",
+        "            HarmCategory.HARM_CATEGORY_HATE_SPEECH: HarmBlockThreshold.BLOCK_LOW_AND_ABOVE,\n",
+        "            HarmCategory.HARM_CATEGORY_SEXUALLY_EXPLICIT: HarmBlockThreshold.BLOCK_LOW_AND_ABOVE,\n",
+        "            HarmCategory.HARM_CATEGORY_DANGEROUS_CONTENT: HarmBlockThreshold.BLOCK_LOW_AND_ABOVE,\n",
+        "        },\n",
+        "    )\n",
+        "    print(\"-- Safety Attributes ---\")\n",
+        "    print(\"\", \"SEVERITY\", \"PROBABILITY\", \"CATEGORY\")\n",
+        "    print(\"-------------------------\")\n",
+        "    for i, score in enumerate(response.candidates[0].safety_ratings):\n",
+        "        if (score.probability_score > probability_threshold) and (\n",
+        "            score.severity_score > severity_threshold\n",
+        "        ):\n",
+        "            style = Back.YELLOW\n",
+        "            valid = False\n",
+        "        else:\n",
+        "            style = Back.WHITE\n",
+        "        print(style, score.severity_score, score.probability_score, score.category)\n",
```

Comment:
For this printing, you might want to make a utility function for it, since it's used in multiple places